From 4aaca54c4188aaedca7d01a28f0c9d11f99d4cb2 Mon Sep 17 00:00:00 2001
From: John McNamara <jmcnamara@cpan.org>
Date: Thu, 3 Aug 2023 01:07:49 +0100
Subject: [PATCH] Fix cmake minizip version check.

Fix Cmake minizip version check that had an incorrect regex and
resulted in the package not being found.

Issue #405
---
 cmake/FindMINIZIP.cmake | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/cmake/FindMINIZIP.cmake b/cmake/FindMINIZIP.cmake
index ed19782e..bef784f5 100644
--- a/cmake/FindMINIZIP.cmake
+++ b/cmake/FindMINIZIP.cmake
@@ -90,9 +90,9 @@ FindStaticLibs(MINIZIP)
 # -------
 
 if(MINIZIP_FOUND)
-   file(STRINGS "${MINIZIP_INCLUDE_DIRS}/zlib.h" MINIZIP_VERSION_CONTENTS REGEX "Version [0-9]+\\.[0-9]+(\\.[0-9]+)?")
-    string(REGEX REPLACE ".*Version ([0-9]+)\\.[0-9]+" "\\1" MINIZIP_VERSION_MAJOR "${MINIZIP_VERSION_CONTENTS}")
-    string(REGEX REPLACE ".*Version [0-9]+\\.([0-9]+)" "\\1" MINIZIP_VERSION_MINOR "${MINIZIP_VERSION_CONTENTS}")
+    file(STRINGS "${MINIZIP_INCLUDE_DIRS}/zlib.h" MINIZIP_VERSION_CONTENTS REGEX "version [0-9]+\\.[0-9]+(\\.[0-9]+)?")
+    string(REGEX REPLACE ".*version ([0-9]+)\\.[0-9]+" "\\1" MINIZIP_VERSION_MAJOR "${MINIZIP_VERSION_CONTENTS}")
+    string(REGEX REPLACE ".*version [0-9]+\\.([0-9]+)" "\\1" MINIZIP_VERSION_MINOR "${MINIZIP_VERSION_CONTENTS}")
     set(MINIZIP_VERSION_PATCH 0)
 
     set(MINIZIP_VERSION_STRING "${MINIZIP_VERSION_MAJOR}.${MINIZIP_VERSION_MINOR}.${MINIZIP_VERSION_PATCH}")

From f477741dd3782101eefb35f9c6f9ed93ee3f642d Mon Sep 17 00:00:00 2001
From: Yihua Liu <yihuajack@live.cn>
Date: Sat, 29 Apr 2023 16:50:51 +0800
Subject: [PATCH] fix: cmake zlib minizip ver required

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e656184d..fb5a2f5e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -219,13 +219,13 @@ enable_language(CXX)
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
 # ZLIB
-find_package(ZLIB REQUIRED "1.0")
+find_package(ZLIB "1.0" REQUIRED)
 list(APPEND LXW_PRIVATE_INCLUDE_DIRS ${ZLIB_INCLUDE_DIRS})
 message("zlib version: " ${ZLIB_VERSION})
 
 # MINIZIP
 if (USE_SYSTEM_MINIZIP)
-    find_package(MINIZIP REQUIRED "1.0")
+    find_package(MINIZIP "1.0" REQUIRED)
     list(APPEND LXW_PRIVATE_INCLUDE_DIRS ${MINIZIP_INCLUDE_DIRS})
 endif()
 
